doc/bucket-types.md

```diff
@@ -0,0 +1,23 @@
+[NooBaa Operator](../README.md) /
+
+# Bucket Types in NooBaa
+NooBaa supports two types of buckets - most commonly referred to as `namespace` and `data`. While all NooBaa buckets are accessible via the S3 API and appear the same to the user, they have different uses and implementations under the hood.
+This document will explaing the meaning of each, as well as the differences between the two.
+
+## Data Buckets
+Data buckets are the 'classic' type of bucket in NooBaa. NooBaa deployments usually create a default backingstore, bucketclass and bucket (`first.bucket`) as part of the product's initialization process. When data is written to these buckets, it is first processed by NooBaa - the data is compressed, deduplicated, encrypted, and split into chunks. These chunks are then stored in the backingstore that the bucket is connected to.
+The only way to access the objects on data buckets is through NooBaa - the chunks cannot be used or deciphered without the system.
+
+## Namespace Buckets
+Namespace buckets work differently than data buckets, since they try¹ to not apply any processing on the objects that are uploaded to them, and act as more of a 'passthrough'.
+The objects can be accessed both from within NooBaa, as well as from whatever storage providers hosts them.
+
+<sup><sub>1. In some cases, object metadata (such as tags) might have to be modified in order to comply with a cloud provider's limits</sub></sup>
```

Comment:
1. would mention the types of NS we supports.
2. Add that when the underlying storage does not support S3 , NooBaa performs the logic and semantics on top what is supported by the underlying storage